test/e2e/framework/util.go

```diff
@@ -1359,14 +1359,14 @@ func podNotPending(c clientset.Interface, podName, namespace string) wait.Condit
 // to terminate or if the pod terminated with an unexpected reason.
 func waitForPodTerminatedInNamespace(c clientset.Interface, podName, reason, namespace string) error {
 	return WaitForPodCondition(c, namespace, podName, "terminated due to deadline exceeded", PodStartTimeout, func(pod *v1.Pod) (bool, error) {
-		if pod.Status.Phase == v1.PodFailed {
-			if pod.Status.Reason == reason {
+		if pod.Status.Phase == v1.PodSucceeded || pod.Status.Phase == v1.PodFailed {
+			if reason == "" ||
+				strings.ToLower(strings.TrimSpace(pod.Status.Reason)) == strings.ToLower(strings.TrimSpace(reason)) {
```

Comment:
I now understand @msau42 's comment. First, I didn't know that a pod won't be deleted (even though the Delete api returns nil error) if it's Status.Phase is `podFailed`. This is probably so the pod can be debugged.

@msau42 's concern is for what could be referred to as a "negative" test case, meaning, calling `waitForPodTerminatedInNamespace` to test that a pod is **not** deleted due to the passed-in reason. Eg. testing that a pod remains due to 1) it's Phase being PodFailed and 2) the reason. In this scenario you need to be able to test that a Failed pod was not deleted and the reason parm can be used to short-circuit the loop.

If we didn't care about this "negative" test case then the `condition` func passed to `WaitForPodCondition` could simply be `func() (bool, error) { return false, nil }`, meaning that `WaitForPodCondition` will loop until pod api `Get` returns an error or it times-out.